homeassistant/components/calendar/__init__.py

```diff
@@ -127,65 +153,11 @@ def state(self):
         if start <= now < end:
             return STATE_ON
 
-        if now >= end:
-            self.cleanup()
-
         return STATE_OFF
 
-    def cleanup(self):
-        """Cleanup any start/end listeners that were setup."""
-        self._cal_data = {
-            'all_day': False,
-            'offset_time': 0,
-            'message': '',
-            'start': None,
-            'end': None,
-            'location': None,
-            'description': None
-        }
```

Comment:
Is it good to make the breaking change and remove the clean up and keep the last event data in state attributes, even when the event has passed?

Reply:
I think it's fine. I think eventually I would want to remove any entity related things from Calendar and just have a calendar specific Lovelace card and automation triggers. Because right now we are not able to deal with overlapping events for example.